src/network/networkd-dhcp-server.c

```diff
@@ -395,3 +404,169 @@ int config_parse_dhcp_server_emit(
                 emit->addresses[emit->n_addresses++] = a.in;
         }
 }
+
+int config_parse_dhcp_static_lease_address(
+        const char *unit,
+        const char *filename,
+        unsigned line,
+        const char *section,
+        unsigned section_line,
+        const char *lvalue,
+        int ltype,
+        const char *rvalue,
+        void *data,
+        void *userdata) {
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        Network *network = data;
+        sd_dhcp_static_lease **current_lease_ref = &network->current_static_lease;
+        sd_dhcp_static_lease *current_lease = NULL;
+        Hashmap **static_leases_ref = &network->dhcp_static_leases;
+        _cleanup_(sd_dhcp_static_lease_unrefp) sd_dhcp_static_lease *old = NULL;
+
+        union in_addr_union addr;
+        int r;
+
+        if (isempty(rvalue)) {
+                return 0;
+        }
+
+        r = in_addr_from_string(AF_INET, rvalue, &addr);
+        if (r < 0) {
+                log_syntax(
+                        unit,
+                        LOG_ERR,
+                        filename,
+                        line,
+                        r,
+                        "Failed to parse DHCPv4 IPv4 address data, ignoring assignment: %s",
+                        rvalue);
+
+                return 0;
+        }
+
+        r = sd_dhcp_static_lease_ensure_allocated(current_lease_ref);
+        if (r < 0)
+                return log_oom();
+        current_lease = *current_lease_ref;
+
+        /* set address */
+        r = sd_dhcp_static_lease_set_address(current_lease, &addr.in);
+        if (r < 0)
+                return log_oom();
+
+        /* if static lease is not complete do nothing */
+        if (!sd_dhcp_static_lease_complete(current_lease))
```

Comment:
In stead of checking the section's completeness here, please verify the section in `network_verify()`.

Reply:
Again this function is not validating the section but checking if the static lease is ready to be put in the `static_leases_ref` hashmap. Are You still sure that I am supposed to use the `network_verify()` function?

Reply:
Each section must be verified after _all_ settings in the section are evaluated. So, please add, say `static_lease_section_verify()`, and call it at `network_verify()`.